v03_pipeline/lib/tasks/reference_data/update_variant_annotations_table_with_updated_reference_dataset.py

```diff
@@ -0,0 +1,50 @@
+import hail as hl
+import luigi
+
+from v03_pipeline.lib.model import ReferenceDatasetCollection
+from v03_pipeline.lib.reference_data.compare_globals import (
+    Globals,
+    get_datasets_to_update,
+)
+from v03_pipeline.lib.tasks.base.base_variant_annotations_table import (
+    BaseVariantAnnotationsTableTask,
+)
+
+
+class UpdateVariantAnnotationsTableWithUpdatedReferenceDataset(
+    BaseVariantAnnotationsTableTask,
+):
+    rdc = luigi.EnumParameter(enum=ReferenceDatasetCollection)
+
+    def complete(self) -> bool:
+        if not super().complete():
+            return False
+
+        annotations_ht_globals = Globals.from_ht(
+            hl.read_table(self.output().path),
+            self.rdc,
+            self.dataset_type,
+        )
+        rdc_ht_globals = Globals.from_ht(
+            self.rdc_annotation_dependencies[f'{self.rdc.value}_ht'],
+            self.rdc,
+            self.dataset_type,
+        )
+        updated_datasets_for_rdc = get_datasets_to_update(
+            self.rdc,
+            annotations_ht_globals,
+            rdc_ht_globals,
+            self.dataset_type,
+        )
+        return len(updated_datasets_for_rdc) == 0
+
+    def update_table(self, ht: hl.Table) -> hl.Table:
+        rdc_ht = self.rdc_annotation_dependencies[f'{self.rdc.value}_ht']
+        rdc_datasets = self.rdc.datasets(self.dataset_type)
```

Comment:
Do we want to touch every `dataset` here or just the ones that require an update?

Reply:
As in... do we want to use: 

```
rdc_datasets = get_datasets_to_update(
             self.rdc,
             annotations_ht_globals,
             rdc_ht_globals,
             self.dataset_type,
         )
```

Reply:
I think I implemented it the way I did because I thought I needed to join every dataset in the rdc - see line 49 `ht = ht.join(rdc_ht, 'left')`..

but can I do `ht = ht.join(rdc_ht[dataset], 'left')`? would it save time/computation to do a join for each dataset that needs an update?

Reply:
Yeah.  I think we only want to re-join against the field that needs it.  It _should_ reduce work!